### PR TITLE
build: replaces tsdx with dts-cli

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

It replaces tsdx which is not maintained anymore with dts-cli.
It also replaces ganache with hardhat for testing contract calls.
Update workflows to use node 18.
Test workflow now tests the SDK for node 16,18 and 20.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: modules/common/package.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
     "node": ">=14"
```
Comment (josemarinas) [inline modules/common/package.json]: Should we unify the required node version?

File: modules/client/test/integration/multisig-client/methods.test.ts [hunk 2]
```diff
@@ -140,7 +131,7 @@ describe("Client Multisig", () => {
       },
     };
     const ipfsUri = await multisigClient.methods.pinMetadata(metadata);
-    const endDate = new Date(Date.now() + 1000 * 60);
+    const endDate = new Date(Date.now() + 1000 * 60 * 60);
```
Comment (josemarinas) [inline modules/client/test/integration/multisig-client/methods.test.ts:134]: You changed endDAte to new Date(0) on the other places, is there a reason why cant do it heere too?

**Review by josemarinas — COMMENTED** [review]
Looks good, I saw that you removed the blockchain server start and close, I guess hardhat does not use that, how does the hardhat package launch the test chain? its done automatically or is there a piece of code that i missed?

If so does the blockchain get reseted on each test?

File: modules/common/package.json (hunk 1 above)
Comment (mathewmeconry) [reply modules/common/package.json]: good idea. done

File: modules/client/test/integration/multisig-client/methods.test.ts (hunk 2 above)
Comment (mathewmeconry) [reply modules/client/test/integration/multisig-client/methods.test.ts:134]: the multisig doesn't support setting the endDate to 0 and choose the end date automatically. So I had to increase it to make sure that it works

**Comment by mathewmeconry** [comment]
> Looks good, I saw that you removed the blockchain server start and close, I guess hardhat does not use that, how does the hardhat package launch the test chain? its done automatically or is there a piece of code that i missed?
> 

It is baked into the `modules/client/test-environment.ts` setup. There it starts hardhat automatically.

> If so does the blockchain get reseted on each test?

No, it doesn't for now we have the same setup where is redeploys the contracts as it was before. But definitely a possibility. We could also activate proper fork testing where we don't need to deploy the contracts anymore.

**Review by josemarinas — APPROVED** [review]

**Pull request merged**